spring-rabbit/src/main/java/org/springframework/amqp/rabbit/AsyncRabbitTemplate.java

```diff
@@ -312,105 +352,132 @@ public RabbitMessageFuture sendAndReceive(String exchange, String routingKey, Me
+	public <C> RabbitConverterFuture<C> convertSendAndReceive(String exchange, String routingKey, Object object,
 			MessagePostProcessor messagePostProcessor) {
 		CorrelationData correlationData = null;
 		if (this.enableConfirms) {
 			correlationData = new CorrelationData(null);
 		}
 		CorrelationMessagePostProcessor<C> correlationPostProcessor = new CorrelationMessagePostProcessor<C>(
 				messagePostProcessor, correlationData);
-		this.template.convertAndSend(exchange, routingKey, message, correlationPostProcessor, correlationData);
+		if (this.container != null) {
+			this.template.convertAndSend(exchange, routingKey, object, correlationPostProcessor, correlationData);
+		}
+		else {
+			MessageConverter converter = this.template.getMessageConverter();
+			if (converter == null) {
+				throw new AmqpIllegalStateException(
+						"No 'messageConverter' specified. Check configuration of RabbitTemplate.");
+			}
+			Message message = converter.toMessage(object, new MessageProperties());
+			correlationPostProcessor.postProcessMessage(message);
+			sendDirect(exchange, routingKey, message, correlationData);
+		}
 		return correlationPostProcessor.getFuture();
 	}
 
+	private void sendDirect(String exchange, String routingKey, Message message,
+			CorrelationData correlationData) {
+		final Channel channel = obtainChannel();
+		message.getMessageProperties().setReplyTo(Address.AMQ_RABBITMQ_REPLY_TO);
+		try {
+			if (channel instanceof PublisherCallbackChannel) {
+				this.template.addListener(channel);
+			}
+			this.template.doSend(channel, exchange, routingKey, message, this.template.isMandatoryFor(message),
+					correlationData);
+		}
+		catch (Exception e) {
+			throw new AmqpException("Failed to send request", e);
+		}
+	}
+
+	private Channel obtainChannel() {
+		DirectReplyToMessageListenerContainer container = null;
+		Channel channel = null;
+		while (this.directConsumers.size() > 0) {
+			container = this.directConsumers.poll();
+			channel = container.getChannel();
+			if (channel != null && channel.isOpen()) {
+				break;
+			}
+		}
+		if (channel == null) {
+			container = new DirectReplyToMessageListenerContainer(this.template.getConnectionFactory());
+			container.setQueueNames(Address.AMQ_RABBITMQ_REPLY_TO);
+			container.start();
+			try {
+				if (!container.isConsuming(10, TimeUnit.SECONDS)) {
+					container.stop();
+					throw new IllegalStateException("ReplyTo container failed to start");
+				}
+			}
+			catch (InterruptedException e) {
+				Thread.currentThread().interrupt();
+				container.stop();
+				throw new IllegalStateException("Interrupted while waiting for ReplyTo container to start");
+			}
+			channel = container.getChannel();
+		}
+		final DirectReplyToMessageListenerContainer theContainer = container;
+		container.setMessageListener(message -> {
+			onMessage(message);
+			try {
+				synchronized (this.directConsumersMonitor) {
+					if (this.running) {
+						this.directConsumers.put(theContainer);
```

Comment:
Because we need a pool of channels - a channel needs to be dedicated for a send/receive operation. I suppose we could implement the pool within the DMLC and use `consumersPerQueue` to increase the pool size I'll take a look at it.